lib/grpnice.gi

```diff
@@ -103,6 +114,9 @@ InstallMethod( GroupByNiceMonomorphism,
 function( nice, grp )
     local   fam,  pre;
 
+    if not IsInjective( nice ) then
+      Error( "<nice> is not injective" );
```

Comment:
```suggestion
      Error( "<nice> is not known to be injective" );
```

Reply:
Well, the current code actually checks injectivity, thus the message is correct.
The question is whether we should instead check `HasIsInjective`, and signal an error if not.